### PR TITLE
date and radio validation

### PR DESCRIPTION
**Pull request by sidgogyan**
### What do you want to achieve?
- date and radio validation

### PR Checklist

- [x] I have tested it locally and all functionalities are working fine.
- [x] I have compared it with mocks and all design elements are the same.
- [ ] I have tested it in IE Browser.

### How was the code tested?

-

### What new thing you came across while writing this code? 
-

### In case you fixed a bug then please describe the root cause of it? 
-

NOTE: Make sure you're comparing your branch with the correct base branch

### CODE REVIEW

File: webplugin/js/app/km-rich-text-event-handler.js [hunk 1]
```diff
@@ -605,6 +609,7 @@ Kommunicate.richMsgEventHandler = {
                     }
                     break;
                 default:
+                    console.log(formElements[i]);
```
Comment (Sanjeevsinghnegi) [inline webplugin/js/app/km-rich-text-event-handler.js:612]: do we require this?

**Review by Sanjeevsinghnegi — CHANGES_REQUESTED** [review]

File: webplugin/js/app/km-rich-text-event-handler.js (hunk 1 above)
Comment (sidgogyan) [reply webplugin/js/app/km-rich-text-event-handler.js]: removed

File: webplugin/js/app/km-rich-text-event-handler.js [hunk 2]
```diff
@@ -576,6 +576,8 @@ Kommunicate.richMsgEventHandler = {
         var name = '';
         var type = '';
         var value = '';
+        let radioerror=true;
```
Comment (Prochnost55) [inline webplugin/js/app/km-rich-text-event-handler.js:579]: we don't use let in web-sdk project 
please change this to var

File: webplugin/js/app/km-rich-text-event-handler.js [hunk 3]
```diff
@@ -576,6 +576,8 @@ Kommunicate.richMsgEventHandler = {
...
+        let idx=-1;
```
Comment (Prochnost55) [inline webplugin/js/app/km-rich-text-event-handler.js:580]: we can change this to index?

**Review by Prochnost55 — CHANGES_REQUESTED** [review]

File: webplugin/js/app/km-rich-text-event-handler.js (hunk 3 above)
Comment (sidgogyan) [reply webplugin/js/app/km-rich-text-event-handler.js]: this is done

**Review by Prochnost55 — DISMISSED** [review]

File: webplugin/js/app/km-rich-text-event-handler.js [hunk 4]
```diff
@@ -635,6 +639,20 @@ Kommunicate.richMsgEventHandler = {
                 postBackData[name] = data[name];
             }
         }
+       if(index!=-1){
+        if (formElements[index].dataset.errorText) {
```
Comment (Sanjeevsinghnegi) [inline webplugin/js/app/km-rich-text-event-handler.js:643]: if (index!=-1 && formElements[index].dataset.errorTex) {}

we can use this

**Review by Sanjeevsinghnegi — CHANGES_REQUESTED** [review]

File: webplugin/js/app/km-rich-text-event-handler.js (hunk 4 above)
Comment (sidgogyan) [reply webplugin/js/app/km-rich-text-event-handler.js]: this is done @Sanjeevsinghnegi

**Review by Sanjeevsinghnegi — APPROVED** [review]

**Review by Prochnost55 — APPROVED** [review]

**Pull request merged**